utils/update-object-and-methods.ts

```diff
@@ -0,0 +1,12 @@
+export function updateObjectAndMethods<T>(
```

Comment:
pthis needs documentation because its super unclear its required for optimistic updates etc to function properly